src/rtt.c

```diff
@@ -411,17 +413,35 @@ void poll_rtt(target *cur_target)
 		if (!rtt_found)
 			/* find rtt control block in target memory */
 			find_rtt(cur_target);
-		/* do rtt i/o if control block found */
 		if (rtt_found) {
-			for (uint32_t i = 0; i < MAX_RTT_CHAN; i++) {
-				rtt_retval v;
-				if (rtt_channel[i].is_enabled && rtt_channel[i].is_configured) {
-					if (rtt_channel[i].is_output)
-						v = print_rtt(cur_target, i);
-					else
-						v = read_rtt(cur_target, i);
-					if (v == RTT_OK) rtt_busy = true;
-					else if (v == RTT_ERR) rtt_err = true;
+			/* check control block not changed or corrupted */
+			if (target_mem_read(cur_target, cblock_header, rtt_cbaddr, sizeof(cblock_header))
+				|| memcmp(saved_cblock_header, cblock_header, sizeof(cblock_header)))
+				rtt_found = false; // force searching control block next poll_rtt()
+		}
+		/* do rtt i/o if control block found */
+		if (rtt_found && rtt_cbaddr) {
+			/* copy control block from target */
+			uint32_t rtt_cblock_size = (sizeof rtt_channel[0]) * (rtt_num_up_chan + rtt_num_down_chan);
```

Comment:
`sizeof(rtt_channel[0])` which also removes the outer parens around the expression